src/vault/use-cases/keys/create_key.go

```diff
@@ -88,37 +88,48 @@ func (uc *createKeyUseCase) Execute(ctx context.Context, namespace, id, algo, cu
 
 func (*createKeyUseCase) eddsaBN254(importedPrivKey string) (eddsa.PrivateKey, error) {
 	if importedPrivKey == "" {
-		key, err := crypto.NewBN254()
+		key, err := crypto2.NewBN254()
 		if err != nil {
-			return key, errors.CryptoOperationError(err.Error())
+			return key, err
 		}
 
 		return key, nil
-	} else {
-		key, err := crypto.ImportBN256(importedPrivKey)
-		if err != nil {
-			return key, errors.InvalidParameterError(err.Error())
-		}
+	}
 
-		return key, nil
+	key := eddsa.PrivateKey{}
+
+	privKeyBytes, err := base64.StdEncoding.DecodeString(importedPrivKey)
+	if err != nil {
+		return key, err
+	}
+
+	_, err = key.SetBytes(privKeyBytes)
+	if err != nil {
+		return key, err
 	}
+
+	return key, nil
 }
 
 func (*createKeyUseCase) ecdsaSecp256k1(importedPrivKey string) (*ecdsa.PrivateKey, error) {
 	if importedPrivKey == "" {
-		key, err := crypto.NewSecp256k1()
+		key, err := crypto.GenerateKey()
 		if err != nil {
-			return key, errors.CryptoOperationError(err.Error())
+			return nil, err
```

Comment:
isn't this a crypto operation error anymore?

Reply:
these errors are not used and they all translate to `InvalidParameterError` in the use case, we can simply return the error.